packages/babel-parser/src/plugins/typescript/scope.js

```diff
@@ -0,0 +1,12 @@
+import type ScopeHandler, { Scope } from "../../util/scope";
+
+interface TypeScriptScope extends Scope {}
+
+const mixinScope = (superClass: Class<Scope>): Class<TypeScriptScope> =>
+  class TypeScriptScope extends superClass {};
```

Comment:
Do you think we will need that? I don't think we will ever have the need of multiple inheritances. as flow and ts will never be used together

I would just do
```
class TypeScriptScope extends Scope {};
```